src/kudu/consensus/raft_consensus.cc

```diff
@@ -3369,6 +3369,19 @@ void RaftConsensus::DoLeaderDetectedCallback() {
   if (ldcb_) ldcb_();
 }
 
+
+Status RaftConsensus::SetCurrentTermBootstrap(int64_t new_term) {
+  LockGuard l(lock_);
+  if (PREDICT_FALSE(new_term <= CurrentTermUnlocked())) {
```

Comment:
Can you explain why this will work?

Since the "consesnsus-metadata" directory is not present on fresh instances, `CurrentTermUnlocked()` will always be `0`, right? So this check will never work if the new term is greater than that.

Reply:
So this check essentially does not allow the term of an entity to never go down. Once set properly on bootstrap, it always will increase either by timing out and doing elections or receiving higher order term from other entities.

Thats why we guard band the caller from passsing in terms that are smaller and equal to Consensus-metadata term.